aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/package.scala

```diff
@@ -0,0 +1,75 @@
+package com.eharmony.aloha.models
+
+import com.eharmony.aloha.dataset.density.Sparse
+
+import scala.collection.{immutable => sci}
+import scala.util.Try
+
+/**
+  * Created by ryan.deak on 8/31/17.
+  */
+package object multilabel {
+
+  // All but the last two types are package private, for testing.  The others are public.
+
+  /**
+    * Features about the input value (NOT including features based on labels).
+    * This should probably be a `sci.IndexedSeq[Sparse]` but `RegressionFeatures`
+    * returns a `collection.IndexedSeq` and using
+    * [[com.eharmony.aloha.models.reg.RegressionFeatures.constructFeatures]] is
+    * preferable and will provide consistent results across many model types.
+    */
+  private[multilabel] type SparseFeatures = IndexedSeq[Sparse]
+
+  /**
+    * Indices of the labels for which predictions should be produced into the
+    * sequence of all labels.  Indices will be sorted in ascending order.
+    */
+  private[multilabel] type LabelIndices = sci.IndexedSeq[Int]
+
+  /**
+    * Labels for which predictions should be produced.  This can be an improper subset of all labels.
+    * `Labels` align with `LabelIndices` meaning `LabelIndices[i]` will give the index of `Labels[i]`
+    * into the sequence of all labels a model knows about.
+    *
+    * @tparam K the type of labels (or classes in the machine learning literature).
+    */
+  private[multilabel] type Labels[K] = sci.IndexedSeq[K]
+
+  /**
+    * Sparse features related to the labels.  Other outer sequence aligns with the `Labels` and `LabelIndices`
+    * sequences meaning `SparseLabelDepFeatures[i]` relates to the features of `Labels[i]`.
+    */
+  private[multilabel] type SparseLabelDepFeatures = Labels[SparseFeatures]
+
+  /**
+    * A sparse multi-label predictor takes:
+    *
+    - features
+    - labels for which a prediction should be produced
```

Comment:
What happens if the labels for which a prediction should be produced contains a label that was not known at train time?